aten/src/ATen/cpu/vec/vec512/vec512_float.h

```diff
@@ -793,6 +793,80 @@ inline void transpose_mxn<float, 16, 16>(
   _mm512_storeu_ps(&dst[15 * ld_dst], p);
 }
 
+// Code referred to FBGEMM:
+// https://github.com/pytorch/FBGEMM/blob/39a423e4ad1a04b77fea81c7d09c3e6f8984fae9/src/UtilsAvx512.cc#L230-L304
+// kernel for transposing mxn where m, n <= 16
+// M + (M + 1) / 2 * 2 + (M + 3) / 4 * 4 + (M + 7) / 8 * 8 + 2 * N instructions
+template <typename T, int M, int N,
+          typename std::enable_if_t<std::is_same<T, float>::value && (M < 16 || N < 16), int> = 0>
+inline void transpose_mxn(const float* src, int64_t ld_src, float* dst, int64_t ld_dst) {
+  // load from src to registers
+  __mmask16 src_mask = (1 << N) - 1;
+  __m512 input[16];
+  int i;
+  for (i = 0; i < M; ++i) {
+    input[i] = _mm512_maskz_loadu_ps(src_mask, &src[i * ld_src]);
+  }
+  for (; i < 16; ++i) {
+    // Not really needed but to avoid uninitialized variable warning.
+    // Shouldn't be much overhead because xor can be executed in parallel with
+    // other instructions.
+    input[i] = _mm512_setzero_ps();
+  }
+
+  // unpacking and interleaving 32-bit elements
```

Comment:
The logic below is same as that in `transpose_mxn<float, 16, 16>`. Can we factor out a common function to share and avoid dup?

Reply:
I removed this template specialization function(`transpose_mxn<float, 16, 16>`).